### ISSUE TITLE
Unable to install from sdist: pyproject.toml [project] requires a name.

### ISSUE THREAD

**Issue by mvaled** [MagicStack/immutables#77]
This started to happen since yesterday in some CI environments using `python:3.10-bullseye` image.  However, the CI jobs using `python:3.8-bullseye` and `python:3.9-bullseye` don't fail.

I've seen this in two different project: one uses `tox` to manage the virtualenv, the other one uses `poetry`.  I don't really know why they are preferring the source dist over the binary wheels. 

```
  Downloading immutables-0.16.tar.gz (84 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 84.5/84.5 KB 12.3 MB/s eta 0:00:00
  Installing build dependencies: started
  Installing build dependencies: finished with status 'done'
  Getting requirements to build wheel: started
  Getting requirements to build wheel: finished with status 'error'
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [1009 lines of output]
      /tmp/pip-build-env-7qjk9iks/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:100: _ExperimentalProjectMetadata: Support for project metadata in `pyproject.toml` is still experimental and may be removed (or change) in future releases.
        warnings.warn(msg, _ExperimentalProjectMetadata)
      configuration error: `project` must contain ['name'] properties
      DESCRIPTION:
          Data structure for the **project** table inside ``pyproject.toml`` (as
          initially defined in :pep:`621`)
      
      GIVEN VALUE:
          {
              "requires-python": ">=3.6"
          }
      
      OFFENDING RULE: 'required'
```

**Comment by chmyaf** [comment]
I have same trouble with python 3.10.2
```
$ pipinstall immutables
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [1007 lines of output]
      /tmp/pip-build-env-ur_i0_t6/overlay/lib/python3.10/site-packages/setuptools/config/pyprojecttoml.py:100: _ExperimentalProjectMetadata: Support for project metadata in `pyproject.toml` is still experimental and may be removed (or change) in future releases.
        warnings.warn(msg, _ExperimentalProjectMetadata)
      configuration error: `project` must contain ['name'] properties
      DESCRIPTION:
          Data structure for the **project** table inside ``pyproject.toml`` (as
          initially defined in :pep:`621`)
```

**Comment by treharne** [comment]
**For those looking for a quick fix** because you're trying to deploy to prod on a Friday night

I forked mvaled 's fix to make sure it doesn't go anywhere. Thank you @mvaled ❤️

You can use that forked version in your **requirements.txt** file like this:

```
# requirements.txt

# ... 
falcon==1.4.1
gunicorn==19.9.0
git+https://github.com/treharne/immutables@avoid-issue-7#egg=immutables
Jinja2==3.0.3
lxml==4.4.1
# ...

```

I will keep this fork available at least until it "just works" without it (i.e. an underlying fix is provided for `immutables` and/or `setuptools` is adjusted), since we are using it in prod!

**Comment by mvaled** [comment]
Hi @treharne PR #78 is now merged and they released 0.17 to pypi (#81).  Even better they are now distributing a wheel for Python 3.10.  I didn't even know they were not building for Python 3.10 before 😂.
